frontend/apps/reader/modules/readerhyphenation.lua

```diff
@@ -100,22 +113,29 @@ function ReaderHyphenation:onPreRenderDocument(config)
     -- This is called after the document has been loaded
     -- so we can use the document language.
 
+    -- Use the one manually set for this document
     local hyph_alg = config:readSetting("hyph_alg")
-    if not hyph_alg then
+    if not hyph_alg then -- If none, use the one manually set as default (with Hold)
+        hyph_alg = G_reader_settings:readSetting("hyph_alg_default")
+    end
+    if not hyph_alg then -- If none, use the one associated with document's language
         hyph_alg = self:getDictForLanguage(self.ui.document:getProps().language)
     end
-    if not hyph_alg then
+    if not hyph_alg then -- If none, use the one manually set as fallback (with Hold)
         hyph_alg = G_reader_settings:readSetting("hyph_alg_fallback")
     end
     if hyph_alg then
         self.ui.document:setHyphDictionary(hyph_alg)
     end
+    -- If we haven't set any, hardcoded English_US_hyphen_(Alan).pdb (in cre.cpp) will be used
     self.hyph_alg = cre.getSelectedHyphDict()
 end
 
-function ReaderHyphenation:onSaveSettings()
-    self.ui.doc_settings:saveSetting("hyph_alg", self.hyph_alg)
-end
+-- We only save hyph_alg when it is manually set, so
+-- that new global fallback or default can be used.
+-- function ReaderHyphenation:onSaveSettings()
+--     self.ui.doc_settings:saveSetting("hyph_alg", self.hyph_alg)
```

Comment:
I don't think this should be removed (or commented out) because these are the per-document settings.